src/components/chatbox/chatbox.tsx

```diff
@@ -0,0 +1,550 @@
+import React, { useEffect, useState, useRef } from 'react'
+
+// https://helios.hashicorp.design/icons/library
+import { IconArrowDownCircle16 } from '@hashicorp/flight-icons/svg-react/arrow-down-circle-16'
+import { IconBulb16 } from '@hashicorp/flight-icons/svg-react/bulb-16'
+import { IconDiscussionCircle16 } from '@hashicorp/flight-icons/svg-react/discussion-circle-16'
+import { IconLoading24 } from '@hashicorp/flight-icons/svg-react/loading-24'
+import { IconSend24 } from '@hashicorp/flight-icons/svg-react/send-24'
+import { IconStopCircle24 } from '@hashicorp/flight-icons/svg-react/stop-circle-24'
+import { IconUser16 } from '@hashicorp/flight-icons/svg-react/user-16'
+import { IconWand24 } from '@hashicorp/flight-icons/svg-react/wand-24'
+// ms is a transient dep
```

Comment:
🤷 💭  `npm ls ms`

<details>
  <summary>Output</summary>

```console
dev-portal@ /Users/kevin/repos/dev-portal
├─┬ @hashicorp/platform-analytics@0.12.0
│ └─┬ @segment/analytics-next@1.51.7
│   └─┬ @segment/tsub@1.0.1
│     └─┬ @stdlib/math-base-special-ldexp@0.0.5
│       └─┬ @stdlib/constants-float64-exponent-bias@0.0.8
│         └─┬ @stdlib/utils-library-manifest@0.0.8
│           └─┬ debug@2.6.9
│             └── ms@2.0.0
├─┬ @hashicorp/platform-cli@2.7.2
│ ├─┬ @typescript-eslint/eslint-plugin@5.54.0
│ │ ├─┬ @typescript-eslint/type-utils@5.54.0
│ │ │ └─┬ debug@4.3.4
│ │ │   └── ms@2.1.2
│ │ └─┬ debug@4.3.4
│ │   └── ms@2.1.2
│ ├─┬ @typescript-eslint/parser@5.54.0
│ │ ├─┬ @typescript-eslint/typescript-estree@5.54.0
│ │ │ └─┬ debug@4.3.4
│ │ │   └── ms@2.1.2
│ │ └─┬ debug@4.3.4
│ │   └── ms@2.1.2
│ └─┬ eslint-config-next@13.2.1
│   ├─┬ eslint-import-resolver-node@0.3.7
│   │ └─┬ debug@3.2.7
│   │   └── ms@2.1.3 deduped
│   ├─┬ eslint-import-resolver-typescript@3.5.3
│   │ └─┬ debug@4.3.4
│   │   └── ms@2.1.2
│   └─┬ eslint-plugin-import@2.27.5
│     ├─┬ debug@3.2.7
│     │ └── ms@2.1.3 deduped
│     └─┬ eslint-module-utils@2.7.4
│       └─┬ debug@3.2.7
│         └── ms@2.1.3 deduped
├─┬ @hashicorp/platform-nextjs-plugin@4.2.0
│ └─┬ @hashicorp/next-optimized-images@0.2.0
│   └─┬ file-loader@3.0.1
│     └─┬ webpack@4.46.0
│       └─┬ micromatch@3.1.10
│         ├─┬ extglob@2.0.4
│         │ └─┬ expand-brackets@2.1.4
│         │   └─┬ debug@2.6.9
│         │     └── ms@2.0.0
│         └─┬ snapdragon@0.8.2
│           └─┬ debug@2.6.9
│             └── ms@2.0.0
├─┬ @hashicorp/sentinel-embedded@0.0.14
│ └─┬ node-flex-serve@0.0.2
│   ├─┬ @shellscape/koa-static@4.0.5
│   │ ├─┬ @shellscape/koa-send@4.1.3
│   │ │ └─┬ debug@2.6.9
│   │ │   └── ms@2.0.0
│   │ └─┬ debug@2.6.9
│   │   └── ms@2.0.0
│   ├─┬ debug@3.2.7
│   │ └── ms@2.1.3 deduped
│   └─┬ koa-router@7.4.0
│     └─┬ debug@3.2.7
│       └── ms@2.1.3 deduped
├─┬ @vercel/fetch@6.2.0
│ ├─┬ @vercel/fetch-retry@5.1.3
│ │ └─┬ debug@4.3.3
│ │   └── ms@2.1.2
│ ├─┬ agentkeepalive@4.2.1
│ │ └─┬ humanize-ms@1.2.1
│ │   └── ms@2.1.3 deduped
│ └─┬ debug@4.3.3
│   └── ms@2.1.2
├─┬ jsonwebtoken@8.5.1
│ └── ms@2.1.3
├─┬ next-remote-watch@2.0.0
│ └─┬ express@4.17.1
│   ├─┬ body-parser@1.19.0
│   │ └─┬ debug@2.6.9
│   │   └── ms@2.0.0
│   ├─┬ debug@2.6.9
│   │ └── ms@2.0.0
│   ├─┬ finalhandler@1.1.2
│   │ └─┬ debug@2.6.9
│   │   └── ms@2.0.0
│   └─┬ send@0.17.1
│     ├─┬ debug@2.6.9
│     │ └── ms@2.0.0
│     └── ms@2.1.1
├─┬ nock@13.2.4
│ └─┬ debug@4.3.2
│   └── ms@2.1.2
└─┬ probe-image-size@7.2.3
  ├─┬ needle@2.9.1
  │ └─┬ debug@3.2.7
  │   └── ms@2.1.3 deduped
  └─┬ stream-parser@0.3.1
    └─┬ debug@2.6.9
      └── ms@2.0.0
```

</details>

Reply:
If we're going to use it, go ahead and declare it as a dependency. Wouldn't want it to disappear if we removed it from another package (also using dependencies like this is incompatible with tools like bun and pnpm since they don't hoist dependencies).

Reply:
done in https://github.com/hashicorp/dev-portal/pull/2128/commits/26f772db305c514e3aefe0321109f48b7a41ef17